src/System.Memory/src/System/Span.Portable.cs

```diff
@@ -305,6 +305,11 @@ public override string ToString()
         {
             if (typeof(T) == typeof(char))
             {
+                object obj = Unsafe.As<Pinnable<T>, object>(ref Unsafe.AsRef(_pinnable));
+                if (obj is string str && _byteOffset == MemoryExtensions.StringAdjustment && _length == str.Length)
+                {
+                    return str;
+                }
```

Comment:
Is this change needed?  When will a `Span<char>` (rather than a `ReadOnlySpan<char>`) wrap a `string`?

Reply:
See code sample here: https://github.com/dotnet/corefx/pull/26880#issuecomment-363632190

Reply:
I personally don't think it's worthwhile to add this check to `Span<char>`.  It's such a corner case to optimize for, and at the expense of adding a type check to other `Span<char>.ToString()` calls.  And in general folks can't depend on ToString producing the same string from multiple calls, so I don't consider the discrepancy a real issue.

Reply:
I don't have a strong opinion on this. I am fine with removing this check for Span. @KrzysztofCwalina, @jkotas - any objections?